cli/cmd/lql.go

```diff
@@ -19,71 +19,217 @@
+		)
+	}
+
+	// run specific flags
+	// env flag to specify a query from disk
+	lqlRunCmd.Flags().BoolVarP(
+		&lqlEnv,
+		"env", "e", false,
+		"run an LQL query by ID (using active profile)",
+	)
+	// start time flag
+	// TODO: come up with reasonable default per UI (1d)
+	lqlRunCmd.Flags().StringVarP(
+		&lqlStart,
+		"start", "", "",
+		"start time for LQL query",
+	)
+	// end time flag
+	// TODO: come up with reasonable default per UI (1d)
+	lqlRunCmd.Flags().StringVarP(
+		&lqlEnd,
+		"end", "", "",
+		"end time for LQL query",
+	)
+	lqlRunCmd.Flags().BoolVarP(
+		&lqlValidate,
+		"validate_only", "", false,
+		"validate query only (do not run)",
 	)
 }
 
-func runLQLQuery(_ *cobra.Command, args []string) error {
-	var query = ""
+// for commands that take a query as input
+func inputQuery(cmd *cobra.Command, args []string) (
+	query string,
+	err error,
+) {
+	var queryID string
 
+	// if an inline argument was provided
+	// determine if it's a query or a query identifier
 	if len(args) != 0 && args[0] != "" {
-		query = args[0]
+		if lqlEnv || lqlRepo {
+			queryID = args[0]
+		} else {
+			query = args[0]
+		}
+	}
+
+	if lqlEnv {
+		var queryResponse api.LQLQueryResponse
+		queryResponse, err = cli.LwApi.LQL.GetQueryByID(queryID)
+		if err == nil && len(queryResponse.Data) != 0 {
+			query = queryResponse.Data[0].QueryText
+		}
+	} else if lqlRepo {
+		err = errors.New("NotImplementedError")
 	} else if lqlFile != "" {
-		lqlQuery, err := ioutil.ReadFile(lqlFile)
+		var fileData []byte
+		fileData, err = ioutil.ReadFile(lqlFile)
 		if err != nil {
-			return errors.Wrap(err, "unable to read file")
+			err = errors.Wrap(err, "unable to read file")
+			return
 		}
-		query = string(lqlQuery)
+		query = string(fileData)
+	} else if lqlURL != "" {
+		msg := "unable to open URL"
+		var response *http.Response
+		var body []byte
+
+		response, err = http.Get(lqlURL)
+		if err != nil {
+			err = errors.Wrap(err, msg)
+			return
+		}
+		defer response.Body.Close()
+
+		if response.StatusCode != 200 {
+			err = errors.Wrap(errors.New(response.Status), msg)
+			return
+		}
+
+		body, err = io.ReadAll(response.Body)
+		if err != nil {
+			err = errors.Wrap(err, msg)
+			return
+		}
+		query = string(body)
 	} else {
-		// avoid asking for a confirmation before launching the editor
+		var firstUseWord string
+		if lqlValidate {
+			firstUseWord = "validate"
+		} else {
+			firstUseWord = strings.Split(cmd.Use, " ")[0]
+		}
 		prompt := &survey.Editor{
-			Message:  "Type an LQL query to run",
+			Message:  fmt.Sprintf("Type a query to %s", firstUseWord),
 			FileName: "query*.sh",
 		}
-		err := survey.AskOne(prompt, &query)
-		if err != nil {
-			return err
-		}
+		err = survey.AskOne(prompt, &query)
 	}
 
-	cli.Log.Debugw("running LQL query", "query", query)
-	response, err := cli.LwApi.LQL.Query(query)
+	return
+}
+
+// standardized cli/error output
+func output(response map[string]interface{}, err error, msg string) error {
```

Comment:
This function was used by everything at one time up until things were broken out and then we layered on both json and human readable formats.  I can just roll this back into the `runQuery()`. 

With respect to human-readable output, what do you propose here?  Asking because `EVENT` is a massive AWS Json blob....

Reply:
Fixed.

Reply:
For now we decided not to worry about tabular output for query run...